WalletWasabi/Tor/TorMonitor.cs

```diff
@@ -144,25 +144,31 @@ await foreach (TorControlReply reply in client.ReadEventsAsync(cancellationToken
 	/// <inheritdoc/>
 	protected override async Task ActionAsync(CancellationToken token)
 	{
-		if (TorHttpPool.TorDoesntWorkSince is { }) // If Tor misbehaves.
+		if (TorHttpPool.TorDoesntWorkSince is not null) // If Tor misbehaves.
 		{
 			TimeSpan torMisbehavedFor = DateTimeOffset.UtcNow - TorHttpPool.TorDoesntWorkSince ?? TimeSpan.Zero;
 
 			if (torMisbehavedFor > CheckIfRunningAfterTorMisbehavedFor)
 			{
 				if (TorHttpPool.LatestTorException is TorConnectCommandFailedException torEx)
 				{
-					if (torEx.RepField == RepField.HostUnreachable)
+					if (torEx.RepField == RepField.OnionServiceIntroFailed)
```

Comment:
Why are we narrowing down the activation of the fallback to specific cases? If there are any problems with the communication just try clearnet and if it is working then we need to fallback.